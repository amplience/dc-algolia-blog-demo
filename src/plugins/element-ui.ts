import Vue from 'vue';
import { Card, Row, Col, Container, Header, Main, Pagination } from 'element-ui';

export default (): void => {
  Vue.use(Card);
  Vue.use(Row);
  Vue.use(Col);
  Vue.use(Container);
  Vue.use(Header);
  Vue.use(Main);
  Vue.use(Pagination);
};
