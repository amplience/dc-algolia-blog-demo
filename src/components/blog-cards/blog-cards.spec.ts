import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import { Col as ElCol, Row as ElRow } from 'element-ui';
import BlogCards from './blog-cards.vue';

describe('BlogCards', () => {
  test('is a BlogCards instance with data for a single Cards', () => {
    const wrapper = shallowMount(BlogCards, {
      stubs: { Card: '<div class="card-stub"></div>', ElCol, ElRow, NuxtLink: RouterLinkStub },
      filters: {
        date: value => value
      },
      propsData: {
        blogArticles: [
          {
            title: 'test-title',
            description: 'test-description',
            link: 'test-id',
            timestamp: '2011-11-18T14:54:39.929Z'
          }
        ]
      }
    });

    expect(wrapper.isVueInstance()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
  });

  test("is a BlogCards instance with data for a multiple Card's", () => {
    const wrapper = shallowMount(BlogCards, {
      stubs: { Card: '<div class="card-stub"></div>', ElCol, ElRow, NuxtLink: RouterLinkStub },
      filters: {
        date: value => value
      },
      propsData: {
        blogArticles: [
          {
            title: 'test-title',
            description: 'test-description',
            link: 'test-id',
            timestamp: '2011-11-17T14:54:39.929Z'
          },
          {
            title: 'test-title 2',
            description: 'test-description 2',
            link: 'test-id2',
            timestamp: '2011-11-18T14:54:39.929Z'
          }
        ]
      }
    });

    expect(wrapper.isVueInstance()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
  });
});
