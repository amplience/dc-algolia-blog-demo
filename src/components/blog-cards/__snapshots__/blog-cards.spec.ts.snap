// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogCards is a BlogCards instance with data for a multiple Card's 1`] = `
<div>
  <div
    class="el-row"
    style="margin-left: -10px; margin-right: -10px;"
  >
    <div
      class="el-col el-col-8"
      style="padding-left: 10px; padding-right: 10px;"
    >
      <div
        class="card-stub"
      />
    </div>
    <div
      class="el-col el-col-8"
      style="padding-left: 10px; padding-right: 10px;"
    >
      <div
        class="card-stub"
      />
    </div>
  </div>
</div>
`;

exports[`BlogCards is a BlogCards instance with data for a single Cards 1`] = `
<div>
  <div
    class="el-row"
    style="margin-left: -10px; margin-right: -10px;"
  >
    <div
      class="el-col el-col-8"
      style="padding-left: 10px; padding-right: 10px;"
    >
      <div
        class="card-stub"
      />
    </div>
  </div>
</div>
`;
